### PR TITLE
jobmanager: fix job is not removed from ack list if it finishes before online

### PR DESCRIPTION
**Pull request by amyangfei**
- A job master could finish before `OnWorkerOnline` is called, in this scenario the job should be removed from wait ack list when `OnWorkerOffline` is called.
- Refine the state transition diagram of job manager

### CODE REVIEW

File: servermaster/job_fsm.go [hunk 1]
```diff
@@ -22,32 +22,45 @@ type jobHolder struct {
...
+//     |                           |   (finish)         |               |
+//     |                           |----------------------------------->|
+//     |                           |                    |               |
+//     | Master                    |                    |               |
+//     |  .OnWorkerOffline         |                    |               |
+//     |  (failover)               |                    |               |
+//     |----------------------------------------------->|               |
+//     |                           |                    |               |
+//     | Master                    |                    |               |
+//     |  .OnWorkerOffline         |                    |               |
```
Comment (maxshuang) [inline servermaster/job_fsm.go:49]: Can the worker check the first heartbeat pong before it do some work?
Seems strange that the worker state is marked finish before it's online.

Comment (amyangfei) [reply servermaster/job_fsm.go:49]: They are separated logic, the fsm can't control whether heartbeat is received.
Besides the worker could finish before any heartbeat is sent.

Comment (nongfushanquan) [reply servermaster/job_fsm.go:49]: this problem should be solved by the worker

Comment (amyangfei) [reply servermaster/job_fsm.go:49]: Worker can try to send one heartbeat during initialization, but we can't ensure the pong message is received, which can infer `OnWorkerOnline` is called in master, this is caused by an underlying assumption: **Worker should continue to work without confirming the heartbeat between master is normal.**
Supposing the following scenario
- During worker initialization, the heartbeat ping message is blocking or heartbeat ping is sent, but heartbeat pong is not received, should the worker continues to work now?

Comment (nongfushanquan) [reply servermaster/job_fsm.go:49]: I think the worker should continue work

**Review by nongfushanquan — APPROVED** [review]
approve this pr ,then we should refine the relationship between the status and heartbeat in the worker

**Pull request merged**